packages/core/test/integration/query-interface/add-show-remove-index.test.ts

```diff
@@ -0,0 +1,304 @@
+
+        expect(indexCheck).to.not.be.undefined;
+        assert(indexCheck);
+        if (['mariadb', 'mysql'].includes(dialect)) {
+          expect(indexCheck.schema).to.be.undefined;
+        } else {
+          expect(indexCheck.schema).to.equal(schema);
+        }
+
+        expect(indexCheck.tableName).to.equal('users');
+        expect(indexCheck.name).to.equal('unique_names');
+        expect(indexCheck.unique).to.equal(true);
+        expect(indexCheck.fields).to.have.length(2);
+        expect(indexCheck.fields[0].name).to.equal('first_name');
+        expect(indexCheck.fields[1].name).to.equal('last_name');
+
+        await queryInterface.removeIndex({ tableName: 'users', schema }, 'unique_names');
+        const indexesAfterRemoval = await sequelize.queryInterface.showIndexes({ tableName: 'users', schema });
+        const indexCheckAfterRemoval = indexesAfterRemoval.find(index => index.name === 'unique_names');
+
+        expect(indexCheckAfterRemoval).to.be.undefined;
+      });
+    });
+  }
+
+  if (sequelize.dialect.supports.index.include) {
+    describe('Indexes with include', () => {
+      beforeEach(async () => {
+        await queryInterface.createTable('users', {
+          id: {
+            type: DataTypes.INTEGER,
+            primaryKey: true,
+            autoIncrement: true,
+          },
+          username: DataTypes.STRING,
+          first_name: DataTypes.STRING,
+          last_name: DataTypes.STRING,
+        });
+      });
+
+      it('should add, show and remove non-unique index with include columns', async () => {
+        try {
+          await queryInterface.addIndex('users', { name: 'user_username', fields: ['username'], include: ['first_name', 'last_name'], unique: false });
+          if (dialect === 'db2') {
+            expect.fail('This should have failed');
+          }
+
+          const indexes = await sequelize.queryInterface.showIndexes('users');
+          const indexCheck = indexes.find(index => index.name === 'user_username');
+
+          expect(indexCheck).to.not.be.undefined;
+          assert(indexCheck);
+          expect(indexCheck.tableName).to.equal('users');
+          expect(indexCheck.name).to.equal('user_username');
+          expect(indexCheck.unique).to.equal(false);
+          expect(indexCheck.fields).to.have.length(1);
+          expect(indexCheck.fields[0].name).to.equal('username');
+          expect(indexCheck.includes).to.have.length(2);
+          expect(indexCheck.includes).to.include('first_name');
+          expect(indexCheck.includes).to.include('last_name');
+
+          await queryInterface.removeIndex('users', 'user_username');
+          const indexesAfterRemoval = await sequelize.queryInterface.showIndexes('users');
+          const indexCheckAfterRemoval = indexesAfterRemoval.find(index => index.name === 'user_username');
+
+          expect(indexCheckAfterRemoval).to.be.undefined;
+        } catch (error) {
```

Comment:
this should use `expect().to.be.rejectedWith`, as db2 must throw

You pretty much never need to use try-catch in a test suite

Reply:
Updated now.